### PR TITLE
refactor(ci): finish matrix migration and update release workflow

### PR DESCRIPTION
**Pull request by gartnera**
# Description

- Migrate remaining e2e tests from advanced to e2e matrix
- Make `workflow_dispatch` actually work
- Use [actions/action-script](https://github.com/actions/github-script) to improve label check logic (shell was doing substring matching not array includes)
- Remove CI checks from release workflow and instead just run all CI on every push to release branches

## Summary by CodeRabbit

- **Chores**
  - Updated workflows to enhance flexibility and coverage of testing, including new test types and refined logic for test execution triggers.
  - Streamlined publish-release workflow, removing redundant checks and tests.
  - Adjusted branch-specific triggers for various workflows.
  - Updated Go version requirement to 1.22 in the documentation.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -27,34 +57,61 @@ jobs:
...
-            echo "DEFAULT_TESTS=true" >> $GITHUB_OUTPUT
-          elif [[ ${{ github.event_name }} == 'schedule' ]]; then
-            echo "UPGRADE_TESTS=true" >> $GITHUB_OUTPUT
-            echo "UPGRADE_LIGHT_TESTS=true" >> $GITHUB_OUTPUT
-            echo "ADMIN_TESTS=true" >> $GITHUB_OUTPUT
-          fi
+        uses: actions/github-script@v7
+        with:
+          script: |
+            console.log(context);
```
Comment (swift1337) [inline .github/workflows/e2e.yml:70]: is it possible to move this file to a separate file? https://github.com/actions/github-script?tab=readme-ov-file#run-a-separate-file

Comment (swift1337) [inline .github/workflows/e2e.yml:70]: Can it contain any secrets?

**Review by swift1337 — APPROVED** [review]
LGTM, left some comments

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (gartnera) [reply .github/workflows/e2e.yml:70]: Yes, but that would require a checkout of the repo which would slow it down a bit.

Comment (gartnera) [reply .github/workflows/e2e.yml:70]: Even if it does the secrets should be redacted

**Review by lumtis — APPROVED** [review]

**Review by CharlieMc0 — APPROVED** [review]

**Pull request merged**